samplekotlin/src/main/java/com/example/chattutorial/ChannelActivity.kt

```diff
@@ -44,33 +43,32 @@ class ChannelActivity : AppCompatActivity(R.layout.activity_channel) {
                 is MessageListViewModel.Mode.Normal -> messageInputViewModel.resetThread()
             }
         }
+
+        messageListViewModel.state.observe(this) {
+            when (it) {
+                is MessageListViewModel.State.NavigateUp -> finish()
+            }
+        }
+
         // step 4 - let the message input know when we are editing a message
-        messageListView.setOnMessageEditHandler {
-            messageInputViewModel.editMessage.postValue(it)
+        messageListView.setOnMessageEditHandler { message ->
+            messageInputViewModel.editMessage.postValue(message)
         }
 
         // step 5 - handle back button behaviour correctly when you're in a thread
-        val backButtonHandler = {
+        channelHeaderView.onBackClick = {
```

Comment:
Cleaned back behavior up a little bit. Didn't need to wrap `backButtonHandler` in a lambda, don't need extra local variable, etc.